replicateexistingcluster/replicateexistingcluster.go

```diff
@@ -0,0 +1,224 @@
+package replicateexistingcluster
+
+//go:generate mockgen -destination=./mock_$GOPACKAGE/export.go . ExportService
+
+import (
+	"context"
+	"encoding/json"
+
+	"golang.org/x/xerrors"
+	corev1 "k8s.io/api/core/v1"
+	schedulingv1 "k8s.io/api/scheduling/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	v1 "k8s.io/client-go/applyconfigurations/core/v1"
+	schedulingcfgv1 "k8s.io/client-go/applyconfigurations/scheduling/v1"
+	cfgstoragev1 "k8s.io/client-go/applyconfigurations/storage/v1"
+
+	"github.com/kubernetes-sigs/kube-scheduler-simulator/export"
+)
+
+// Service has two exportServices.
+// existingClusterExportService is used to export resources from an existing cluster.
+// simulatorExportService is used to import(replicate) the resources to the simulator.
+type Service struct {
+	existingClusterExportService ExportService
+	simulatorExportService       ExportService
+}
+
+type ExportService interface {
+	Export(ctx context.Context, opts ...export.Option) (*export.ResourcesForExport, error)
+	Import(ctx context.Context, resources *export.ResourcesForImport, opts ...export.Option) error
+	IgnoreErr() export.Option
+	IgnoreSchedulerConfiguration() export.Option
+}
+
+// NewReplicateExistingClusterService initializes Service.
+func NewReplicateExistingClusterService(exportService ExportService, existingClusterExportService ExportService) *Service {
+	return &Service{
+		existingClusterExportService: existingClusterExportService,
+		simulatorExportService:       exportService,
+	}
+}
+
+// ImportFromExistingCluster gets resources from existing cluster via existingClusterExportService
+// and then apply those resources to the simulator.
+// Note: this method doesn't handle scheduler configuration.
+// If users want to use their scheduler configuration, they need to use `KUBE_SCHEDULER_CONFIG_PATH` env.
+func (s *Service) ImportFromExistingCluster() error {
+	ctx := context.Background()
+	expRes, err := s.existingClusterExportService.Export(ctx)
+	if err != nil {
+		return xerrors.Errorf("call Export of existingClusterExportService: %w", err)
+	}
+	impRes, err := cnvtToResourcesForImportFromResourcesForExport(expRes)
+	if err != nil {
+		return xerrors.Errorf("call cnvtToResourcesForImportFromResourcesForExport: %w", err)
+	}
+	// Import to the simulator.
+	if err := s.simulatorExportService.Import(ctx, impRes, s.simulatorExportService.IgnoreErr(), s.simulatorExportService.IgnoreSchedulerConfiguration()); err != nil {
+		return xerrors.Errorf("call Import of the simulater export service: %w", err)
+	}
+	return nil
+}
+
+func cnvtToResourcesForImportFromResourcesForExport(expRes *export.ResourcesForExport) (*export.ResourcesForImport, error) {
```

Comment:
I think these convert logic should be provided in `export` package.
Maybe, you might want to create a new file `util.go` in this package which includes all utility funcs for export service.
https://github.com/kubernetes-sigs/kube-scheduler-simulator/tree/master/export

Reply:
Thanks, good idea!